File: PoSBackend/ViewModels/Receipt.cs
using System.ComponentModel.DataAnnotations;

namespace PoSBackend.ViewModels
{
    public class Receipt
    {
        [Required]
        public string Date { get; set; } = string.Empty;
        [Required]
        public decimal? Total_amount { get; set; }
        [Required]
        public decimal? Total_discount_amount { get; set; }
        [Required]
        public decimal? Subtotal { get; set; }
        [Required]
        public decimal? Trade_discount { get; set; }
        [Required]
        public decimal? Grand_total { get; set; }
        [Required]
        public List<Order> Orders { get; set; } = new List<Order>();
    }

    public class ReceiptViewModel
    {
        public int Id { get; set; }
        public string Code { get; set; } = string.Empty;
        public string Date { get; set; } = string.Empty;
        public decimal Grand_total { get; set; }
    }

    public class ReceiptDetailViewModel
    {
        public int Id { get; set; }
        public string Code { get; set; } = string.Empty;
        public string Date { get; set; } = string.Empty;
        public decimal Total_amount { get; set; }
        public decimal Total_discount_amount { get; set; }
        public decimal Subtotal { get; set; }
        public decimal Trade_discount { get; set; }
        public decimal Grand_total { get; set; }
        public List<OrderViewModel> Orders { get; set; } = new List<OrderViewModel>();
    }
}

File: PoS2Vue/src/interfaces/Receipt.ts
import type { Order, OrderModel } from "./Order";

export interface Receipt {
    id: number
    code: string
    date: string
    grand_total: number
}

export interface ReceiptDetail {
    id: number
    code: string
    date: string
    total_amount: number
    total_discount_amount: number
    subtotal: number
    trade_discount: number
    grand_total: number
    orders: Order[]
}

export interface ReceiptModel {
    date: string
    total_amount: number
    total_discount_amount: number
    subtotal: number
    trade_discount: number
    grand_total: number
    orders: OrderModel[]
}